src/components/dashboard/ClaimButton.js

```diff
@@ -87,12 +90,69 @@ const ClaimButton = ({ isCitizen, entitlement, nextClaim, loading, onPress, styl
   </CustomButton>
 )
 
+const ClaimAnimationButton = props => {
```

Comment:
looks  very unclear.
by my opinion better to do as the following:
1. after each render store previous entitlement, nextClaim values
2. if entitlement still zero - just render button with props passed (countown will be shown)
3. if entitlement == 0 -- rendering card with 2 buttons (first one with previous entitlement/ nextClaim values). After render starting aniumation

```
const ClaimAnimationButton = ({ styles, entitlement, nextClaim, onPress, ...buttonProps }) => {
  const cardRef = useRef()
  const prevEntitlement = useRef()
  const prevClaimInterval = useRef()

  const setCardRef = useCallback(ref => cardRef.current = ref, []);

  useEffect(() => {
    prevEntitlement.current = entitlement
    prevClaimInterval.current = nextClaim

    if (entitlement) {
      cardRef.current.flip()
    }
  }, [entitlement, nextClaim])

  if (!entitlement) {
    return (
      <ClaimButton {...buttonProps} entitlement={entitlement} nextClaim={nextClaim}  />
    )
  }

  return (
    <CardFlip
      style={styles.cardContainer}
      ref={setCardRef}
      flipDirection="x"
      duration={1000}
      perspective={1000}
    >
      <ClaimButton {...buttonProps} entitlement={prevEntitlement.current} nextClaim={prevClaimInterval.current} />
      <ClaimButton {...buttonProps} entitlement={entitlement} nextClaim={nextClaim} onPress={onPress} />
    </CardFlip>
  )
}
```